app/controllers/cart_placements_controller.rb

```diff
@@ -0,0 +1,33 @@
+class CartPlacementsController < ApplicationController
+    def new 
+        @cart_placement = CartPlacement.new
+    end
+    
+    def index
+        @cart_placement = CartPlacement.new
+    end
+    
+    def create
+        @cart = current_cart
+        @placement = @cart.cart_placements.new(placement_params)
+        @cart.save
+        @placement.save
+        session[:cart_id] = @cart.id
```

Comment:
I don't know if you are still using session here -- I thought you would rely on the database.  It seems extra.

Reply:
This controller isn't being used, I just forgot to destroy it before.